paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -201,6 +200,31 @@ internal class PaymentSheetViewModel @Inject internal constructor(
 
     private val confirmationHandler = confirmationHandlerFactory.create(viewModelScope)
 
+    private val isAwaitingEagerLaunchResult = MutableStateFlow<Boolean?>(null)
+
+    internal val contentVisible: StateFlow<Boolean> = combineAsStateFlow(
+        isAwaitingEagerLaunchResult,
+        confirmationHandler.state.mapAsStateFlow { contentVisibleFromConfirmationState(it) }
+    ) { isAwaitingEagerLaunchResult, contentVisibleFromConfirmationState ->
+        isAwaitingEagerLaunchResult == false && contentVisibleFromConfirmationState
+    }
+
+    private fun contentVisibleFromConfirmationState(state: ConfirmationHandler.State) =
```

Comment:
I'd add some comments here, clarifying that we want the content underneath bottomsheet based payment method views to be hidden to avoid weird visual overlap effects.